src/Specification.cpp

```diff
@@ -60,6 +60,7 @@ const std::string& Specification::fileExtension( void )
 Specification::Specification( void )
 : m_asmType( AsmType::SYNCHRONOUS )
 , m_name()
+, m_location( "", "", "", "", "" )
```

Comment:
Please add an default ctor to URI

Reply:
OK, I've a better solution to this problem. Since this is a violation of the `URI` definition anyway, I'll exchange the `m_location` with a `::Ptr` just like the other members of `Specification`.